### ISSUE TITLE
Can dosbox-x running on Windows mount a network Samba share served by Linux?

### ISSUE THREAD

**Issue by FlexMcMurphy** [joncampbell123/dosbox-x#3339]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

Hello,

I am not able to try myself and to confirm this at the moment but just wanted to check please.

I want to run dosbox-x from Windows and use it to run a 16-bit DOS application that will reside on a Linux Samba share and need that DOS application to be accessible from several Windows clients each running dosbox-x.

I have a drive attached to a Raspberry Pi running a Linux OS (OSMC). I use Samba server to access that drive around my LAN.
For instance the share appears in my Windows 10 Network Neighborhood or I can also "Map Network Drive" in Windows and give it a drive letter.
I tried with FreeDOS and it seemed incredibly complicated to set up access to that network share.

With Dosbox-x is it really as simple as:

`mount g: \\servername\path`

Because from reading the wiki it says:
`Support for direct browsing for UNC network paths (Windows only)
With this feature you can access or browse UNC network paths in the form \\COMPUTER\FOLDER directly on Windows systems...`

Does it matter that the network path points to a share served up by a `Linux OS`?

Why is this so much more complicated on FreeDOS? You have to install network support yourself and configure it.
So I can't believe it's really so easy in dosbox-x.... does it have a samba client? What SMB protocol does it use to connect to Samba shares?
Don't you need to make any changes to authentication protocols from Windows or maybe enable SMBv1?

Thank you,

Flex

**Comment by rderooy** [comment]
@FlexMcMurphy if your on a Windows host, you should be able to mount or access a UNC path **IF** you can likewise access it from the Windows Command Prompt.

e.g., if your able to do something like `dir \\servername\path` from the windows command prompt, you should be able to do the same from the DOSBox-X prompt. This should work with pretty much any command, including `mount`, `dir` and `copy` on the DOSBox-X prompt. e.g., `copy \\servername\path\file .`

Note: this will not work on other platforms like Linux or macOS as it relies on the Windows hosts CIFS/SMB support, DOSBox-X just has the host handle it.

**Comment by FlexMcMurphy** [comment]
Hi @rderooy 
Thanks for responding. I wasn't sure from the Wiki if the network share HAD to be on a Windows machine or if it didn't matter as long as the HOST running dosbos-x was a Windows machine.

So it probably will work then since my host is Windows 10 and the Samba shares, being hosted by a Linux OS, are already appearing in my Windows File Explorer Network Neighborhood window.

I'll report back when I get a chance to give that a try!

> Note: this will not work on other platforms like Linux or macOS as it relies on the Windows hosts CIFS/SMB support, DOSBox-X just has the host handle it.

So does this mean that dosbox-x works with the latest Samba protocol version? Because I didn't like that about FreeDOS which only seemed compatible with the most outdated and insecure Samba protocol and authentication standards.

Cheers,
Flex

**Comment by rderooy** [comment]
Correct. DOSBox-X has no CIFS/SMB support, it is handled by the host, so as long as the host supports it, it should work.

**Comment by arrowgent** [comment]
make sure to allow accessing those ports on the raspberry pi linux firewall
and i would limit access to only your local intranet subnet...
